### PR TITLE
Allow use of a RepositoryRef when creating a CodeCommitSoureAction

### PR DESCRIPTION
**Pull request by RomainMuller**
When trying to use an imported CodeCommit repository (e.g. from a separate stack), one needs ot be able to create the source action using a `RepositoryRef` and not a `Repository`. In order for this to work, the repository clone URLs needed to be made available on the `RepositoryRef` class.

### CODE REVIEW

File: packages/@aws-cdk/aws-codecommit/lib/repository.ts [hunk 1]
```diff
@@ -182,6 +186,24 @@ class ImportedRepositoryRef extends RepositoryRef {
...
+    }
+
+    public get repositoryCloneUrlSsh() {
+        return new cdk.FnConcat('ssh://git-codecommit.',
+                                new cdk.AwsRegion(),
+                                '.',
+                                new cdk.AwsURLSuffix(),
+                                '/v1/repos/',
+                                this.repositoryName).toString();
+    }
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codecommit/lib/repository.ts]: There seems to be a fair amount of duplication between the 2 methods - does extracting a third, private one, make sense here?

**Comment by RomainMuller** [comment]
I include `package-lock.json` changes whenever they pop in (that's my passive-aggressive way to deal with them). I can however make sure they're separated into their own commit, though.

File: packages/@aws-cdk/aws-codecommit/lib/repository.ts (hunk 1 above)
Comment (RomainMuller) [reply packages/@aws-cdk/aws-codecommit/lib/repository.ts]: That's fair, let me do that...

**Comment by RomainMuller** [comment]
Now devoid of `package-lock.json` updates!

**Comment by RomainMuller** [comment]
Oh right, I got distracted by the fact GitHub won't show those... I'll prune them before merging.

**Pull request merged**